sky/engine/core/script/dart_controller.cc

```diff
@@ -68,6 +68,16 @@ DartController::~DartController() {
   if (dom_dart_state_) {
     // Don't use a DartIsolateScope here since we never exit the isolate.
     Dart_EnterIsolate(dom_dart_state_->isolate());
+
+    // Run the framework's shutdown handler.
+    {
+      DartApiScope dart_api_scope;
+      Dart_Handle ui_library = Dart_LookupLibrary(ToDart("dart:ui"));
+      if (!LogIfError(ui_library)) {
+        DartInvokeField(ui_library, "_onShutdown", {});
+      }
+    }
```

Comment:
> Would it be fair to say that if we reach this point it means an improper shutdown occurred?

Reached which point?

The line of code this comment is attached to is executed in the normal shutdown sequence.  It's the point just before we spin down the Dart isolate for a particular view.